### PR TITLE
[add] content switching logic to app

### PR DESCRIPTION
**Pull request by Way2nnadi**
- The nav bar and the footer are now fixed and only the content body is swapped in and out
- Clean up improper `constants.js` file placement
- rename `userPage` folder to `userProfile`
- minor css clean up for `UserProfile` component

### CODE REVIEW

File: src/components/app/components/AppLink.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+const AppLink = ({ to, children }) => <NavLink to={to}>{children}</NavLink>;
```
Comment (joshuaobinna) [inline src/components/app/components/AppLink.js]: i dont understand what's going on here

Comment (joshuaobinna) [inline src/components/app/components/AppLink.js]: what is happening here

File: src/components/app/components/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+const UserMenu = (
+    <Menu>
+        <Item key={USER_PROFILE}>
+            <AppLink to={PROFILE} key={PROFILE}>{USER_PROFILE}</AppLink>
+        </Item>
+        <Item key={SIGN_OUT}>{SIGN_OUT}</Item>
+    </Menu>
+);
+
+const items = [
```
Comment (joshuaobinna) [inline src/components/app/components/Navbar.js]: is this an array of components?

Comment (joshuaobinna) [inline src/components/app/components/Navbar.js]: is lines 32 to 40 an array of components or what?

Comment (joshuaobinna) [inline src/components/app/components/Navbar.js]: is lines 32 to 40 a array of components or what?

File: src/components/home/index.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
-import components from './components';
+import * as components from './components';
```
Comment (joshuaobinna) [inline src/components/home/index.js]: is ' * ' a wild card? what is it doing here
line 1

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "agroeasy",
-  "version": "1.1.0",
+  "version": "1.1.2",
```
Comment (Simpcyclassy) [inline package.json]: Why did you change the version from 1.1.1 in the previous PR to 1.1.2?

File: src/components/app/components/Footer.js [hunk 5]
```diff
@@ -1,19 +1,15 @@
...
 
 const footers = FOOTER_DATA.map(footer => (
     <div className={DYNAMIC_FOOTER} key={footer.key}>
         <h3>{footer.header}</h3>
-        {
-            footer.items.map(item => (
-                <div key={item.title}>{item.title}</div>
-            ))
-        }
+        {footer.items.map(item => <div key={item.title}>{item.title}</div>)}
```
Comment (Simpcyclassy) [inline src/components/app/components/Footer.js]: Why did you make this one line of code

File: src/components/app/components/Navbar.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import React from 'react';
+import { Avatar,  Dropdown, Layout, Menu } from 'antd';
+
+import AppLink from './AppLink';
+import home from '../../home';
+
+import signin from '../../signin';
+import signup from '../../signup';
+import { LOGO, MARKET_TEXT, NAVBAR, PATHS, USER_AVATAR } from '../constants';
+
```
Comment (Simpcyclassy) [inline src/components/app/components/Navbar.js]: Why did you separate the imports being that they are parts of the app

File: src/components/app/components/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            <Header className={MAIN_NAV}>
+                <Menu
+                    className={NAV_MENU}
+                    mode={NAV_MODE}
+                    theme={NAV_THEME}
+                >
+                    {
+                        items.map(item => {
+                            const { key } = item;
+                            const className = key === AVATAR ? CONTAINER : '';
```
Comment (Simpcyclassy) [inline src/components/app/components/Navbar.js]: I don't understand the logic behind this statement

File: package.json (hunk 4 above)
Comment (Way2nnadi) [reply package.json]: So this is called [semantic versioning](https://semver.org/). It lets developers know that a change has been made to the code base. The changes that I am making right now are typed to version 1.1.2. Next week, I will introduce semantic version to you all. It is incredibly important but simple to understand.

Comment (Way2nnadi) [reply package.json]: @Simpcyclassy good job in spotting this

File: src/components/app/components/AppLink.js (hunk 1 above)
Comment (Way2nnadi) [reply src/components/app/components/AppLink.js]: this is javascript destructing of the argument. It is the same as:
```
const AppLink = prop => {
    const { to, children } = prop;
    return <NavLink to={to}>{children}</NavLink>;
}
```

File: src/components/app/components/Footer.js (hunk 5 above)
Comment (Way2nnadi) [reply src/components/app/components/Footer.js]: if it can fit on one line, make it fit on one line, especially if it doesn't make the code harder to read. But that a balance that you have to strike for yourself. I personally prefer one lines because I can read it like english.

File: src/components/app/components/Navbar.js (hunk 6 above)
Comment (Way2nnadi) [reply src/components/app/components/Navbar.js]: You want to separate the imports of external dependencies from those of internal files. We will go over this during our next meeting.

File: src/components/app/components/Navbar.js (hunk 2 above)
Comment (Way2nnadi) [reply src/components/app/components/Navbar.js]: Correct, it is an array of components

File: src/components/app/components/Navbar.js (hunk 7 above)
Comment (Way2nnadi) [reply src/components/app/components/Navbar.js]: `line 32-40` we are defining our items (array of components). Each one has a key value assigned to it. This logic as intended to assign a specific class value to the avatar menu item given that it is different from the rest of the items in the menu. `key === AVATAR ? CONTAINER : '';` is a [ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

File: src/components/home/index.js (hunk 3 above)
Comment (Way2nnadi) [reply src/components/home/index.js]: Yes is a wild card, that says that all the imports from a specific value and assign it a name. Read over [es6 imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to understand what is happening here

File: package.json (hunk 4 above)
Comment (Simpcyclassy) [reply package.json]: OK. That's cool

File: src/components/app/components/Footer.js (hunk 5 above)
Comment (Simpcyclassy) [reply src/components/app/components/Footer.js]: OK

File: src/components/app/components/Navbar.js (hunk 6 above)
Comment (Simpcyclassy) [reply src/components/app/components/Navbar.js]: OK

File: src/components/app/components/Navbar.js (hunk 7 above)
Comment (Simpcyclassy) [reply src/components/app/components/Navbar.js]: OK. I actually knew it is a ternary. I am a little confused what line 58 ought to do exactly being that AVATAR and CONTAINER are strings

File: src/components/app/components/AppLink.js (hunk 1 above)
Comment (joshuaobinna) [reply src/components/app/components/AppLink.js]: ok

File: src/components/app/components/Navbar.js (hunk 2 above)
Comment (joshuaobinna) [reply src/components/app/components/Navbar.js]: ok

File: package.json (hunk 4 above)
Comment (joshuaobinna) [reply package.json]: thats my girl

File: src/components/app/components/Footer.js (hunk 5 above)
Comment (joshuaobinna) [reply src/components/app/components/Footer.js]: me too
i prefer one line code

File: src/components/app/components/Navbar.js (hunk 7 above)
Comment (Way2nnadi) [reply src/components/app/components/Navbar.js]: `AVATAR` and `CONTAINER` are variables that are holding strings. They are not strings by default. I am setting `className` to the value that `CONTAINER` is holding is the value if `key` is strictly equal to `AVATAR` else set `className` to an empty string

Comment (EngrEric) [reply src/components/app/components/Navbar.js]: in simple term, you are saying.... hey key are you in all conditions equal to `AVATAR`?
if your answer is yes, return `CONTAINER`, if your answer is no, return `""` (which is empty string).
From this logic, i learnt that we have filtered the class value of the avatar menu item from the rest of the items in the menu.

Comment (Simpcyclassy) [reply src/components/app/components/Navbar.js]: OK!

Comment (Way2nnadi) [reply src/components/app/components/Navbar.js]: @EngrEric thats exactly correct

**Comment by Simpcyclassy** [comment]
Looks good to me
